prometheus.lua

```diff
@@ -76,6 +76,66 @@ local DEFAULT_BUCKETS = {0.005, 0.01, 0.02, 0.03, 0.05, 0.075, 0.1, 0.2, 0.3,
 -- Prefix for internal shared dictionary items.
 local KEY_INDEX_PREFIX = "__ngx_prom__"
 
+-- Accept range for second byte of utf8
+local accept_range = {
+  {lo = 0x80, hi = 0xBF},
+  {lo = 0xA0, hi = 0xBF},
+  {lo = 0x80, hi = 0x9F},
+  {lo = 0x90, hi = 0xBF},
+  {lo = 0x80, hi = 0x8F}
+}
+
+-- Validate utf8 string for label values.
+--
+-- Args:
+--   str: string
+-- Returns:
+--   (bool) whether the input string is a valid utf8 string.
```

Comment:
What I'd like to suggest is to provide the same API as utf8_validator and return the position of the first non-valid byte as the second return value. This will allow us to use the valid leading portion of a string as a value instead of defaulting to the empty string.

Something like this:
```lua
valid, idx = validate_utf8_string(label_values[idx])
if not valid then
  label_value = string.sub(label_values[idx], 1, idx-1)
end
```

This will also allow you to offer a performance improvement PR to the utf8_validator.lua library.

What do you think?

Reply:
Agree, let me do it.